pkg/reconciler/autoscaling/hpa/resources/keda.go

```diff
@@ -128,6 +131,20 @@ func DesiredScaledObject(ctx context.Context, pa *autoscalingv1alpha1.PodAutosca
 				return nil, fmt.Errorf("query is missing for custom metric: %w", err)
 			}
 
+			tmpl, err := template.New("query").Funcs(sprig.TxtFuncMap()).Parse(query)
+			if err != nil {
+				return nil, fmt.Errorf("template initialization failed: %w", err)
+			}
+			values := map[string]string{
+				"revisionName": pa.Name,
```

Comment:
As a side note Serving supports byo revision names instead of being auto-generated. 
```
kind: Service
metadata:
  name: hello
spec:
  template:
    metadata:
      name: hello-1
```
If you want to change that service then you need to manually change `hello-1` to something else. PA name is always the same as with revision name.

Reply:
Thanks! I think it would be helpful to include this, along with the other points we've discussed, in a "Usage Recommendations" document. Should we log an issue to capture these notes? That way, we can reference it and update it with new considerations as they come up.

Reply:
issue #86